test/aqua/test_phase_estimator.py

```diff
@@ -0,0 +1,200 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test phase estimation"""
+
+import unittest
+from test.aqua import QiskitAquaTestCase
+import numpy as np
+from qiskit.aqua.algorithms.phase_estimators import PhaseEstimator, HamiltonianPE
+from qiskit.aqua.algorithms.phase_estimators.hamiltonian_pe import TempPauliEvolve
+from qiskit.aqua.operators.evolutions import PauliTrotterEvolution
+import qiskit
+from qiskit.aqua.operators import (H, X, Y, Z)
+
+
+class TestHamiltonianPE(QiskitAquaTestCase):
+    """Tests for obtaining eigenvalues from phase estimation"""
+
+    # pylint: disable=invalid-name
+    def hamiltonian_pe(self, hamiltonian, state_preparation=None, num_evaluation_qubits=10,
+                       backend=qiskit.Aer.get_backend('qasm_simulator')):
+        """Run HamiltonianPE and return result with all  phases."""
+        qi = qiskit.aqua.QuantumInstance(backend=backend, shots=100000)
+        phase_est = HamiltonianPE(
+            num_evaluation_qubits=num_evaluation_qubits,
+            hamiltonian=hamiltonian, quantum_instance=qi,
+            state_preparation=state_preparation, evolution=TempPauliEvolve())
+        result = phase_est.run()
+        return result
+
+    # pylint: disable=invalid-name
+    def test_pauli_sum_1(self):
+        """Two eigenvalues from Pauli sum with X, Z"""
+        a1 = 0.5
+        a2 = 1.0
+        hamiltonian = (a1 * X) + (a2 * Z)
+        state_preparation = H.to_circuit()
+        result = self.hamiltonian_pe(hamiltonian, state_preparation)
+        phase_dict = result.filter_phases(0.162, as_float=True)
+        phases = list(phase_dict.keys())
+        self.assertAlmostEqual(phases[0], 1.119, delta=0.001)
+        self.assertAlmostEqual(phases[1], -1.119, delta=0.001)
+
+    # pylint: disable=invalid-name
```

Comment:
This method can probably be removed because it is just a one liner; `hamiltonian = (0.5 * X) + Y + Z`. 
If you don't want to define it inside the tests you can instead put it in the `setUp` method, that's how it's usually done:
```python
def setUp(self):
    super().setUp()
    self.hamiltonian = (0.5 * X) + Y + Z

# then use self.hamiltonian in other methods
```